### ISSUE TITLE
locales fetched from url in defineI18nLocale not working after nuxi generate

### ISSUE THREAD

**Issue by fcieminski** [nuxt-modules/i18n#2156]
### Environment

Nuxi 3.5.3                
Nuxt project info:                                                                                                                                                                20:20:10

------------------------------
- Operating System: Windows_NT
- Node Version:     v18.15.0
- Nuxt Version:     3.5.3
- Nitro Version:    2.4.1
- Package Manager:  yarn@1.22.17
- Builder:          vite
- User Config:      modules, ssr, app, css, nitro, image, runtimeConfig, alias, i18n, gtag
- Runtime Modules:  @nuxtjs/i18n@8.0.0-beta.12-28110710.9b54123, @nuxt/image@1.0.0-rc.1, nuxt-gtag@0.5.7, @vee-validate/nuxt@4.9.6
- Build Modules:    -

### Reproduction

Because it's happens only on generated app, there's no reproduction. But I'll precisly describe the bug.

### Describe the bug

While playing with the development environment, all the locales fetched by defineI18nLocale work fine. I can change my locale through switchLocalePath in my language menu, and the changes are propagated instantly. However, the problem arises with the generated build for static production when ssr: false is used. The default language of my page is not respected, and none of the options inside the language selector menu work. Manually adding a prefix to the URL also does not work. The app remains in English.

Here's my nuxt config:
```
i18n: {
    experimental: {
      jsTsFormatResource: true
    },
    defaultLocale: 'pl',
    locales: [
      {
        code: 'en',
        name: 'english',
        file: 'locale.ts'
      },
      {
        code: 'pl',
        name: 'polish',
        file: 'locale.ts'
      },
      {
        code: 'ua',
        name: 'український',
        file: 'locale.ts'
      },
    ],
    langDir: 'locales',
    precompile: {
      strictMessage: false
    }
  },
```

Here's how my `defineI18nLocale` looks like:
```
import { MAPPED_LOCALES } from '@/enums'
import { Locale, StrapiResponse } from '@/types'

export default defineI18nLocale(async (locale: keyof typeof MAPPED_LOCALES) => {
  const currentLocale = MAPPED_LOCALES[locale]

  const { data } = await $fetch<StrapiResponse<Locale>>(`URL?locale=${currentLocale}`, {
    headers: {
      Authorization: 'AUTH'
    }
  })

  return data.attributes.content
})
```
To sum up, everythings work great on dev, but not after nuxi generate/nuxi preview, it's like locales are not fetched during generate proces.

**Comment by kazupon** [comment]
Thank you for your reporting!

I appreciate your issue reporting info.
But, We need minimal reproduction with stackbliz or GitHub repo, to resolve your issue.
Could you give us it please?

**Comment by fcieminski** [comment]
@kazupon here's stackblitz example - https://stackblitz.com/edit/github-2c5yt2-3hxgw4?file=nuxt.config.ts but remember, you have to generate and preview the project to see the issue.
